swimlane/core/cursor.py

```diff
@@ -58,8 +56,9 @@ def _evaluate(self):
                         break
 
                 if any([
-                    len(raw_elements) < self.page_size,
-                    (self.__limit and len(self._elements) >= self.__limit)
+                    (len(raw_elements) < self.page_size),
+                    (self.__limit and len(self._elements) >= self.__limit),
+                    (self.page_size == 0)
```

Comment:
I don't see us decrementing page_size anywhere. If you wanted to put gate condition, that the place for it will be higher, I would also check if we accept page size 0 in reports to indicate unlimited page size, I don't think we do, but check 0 might be a valid value.

Reply:
I did test out page size 0 and it got back with all the records I was expecting. However, without that check there it only asks if the number of results is less than the page size (we never get less than 0 results) or greater than the limit if there is one (which may also never happen). Adding this condition prevents that infinite loop where it keeps asking for more records but there aren't any.